### ISSUE TITLE
[magit]  the K is bound to magit-file-untrack ????

### ISSUE THREAD

**Issue by zilongshanren** [syl20bnr/spacemacs#3362]
Shouldn't it be bound to magit-discard?  @syl20bnr  Anything missing ?

I'm on the latest develop branch.

**Comment by zilongshanren** [comment]
@TheBB  Any clue? Thanks.

**Comment by CestDiego** [comment]
I have discard when it's on a region, not on a file. And also can confirm that I can't use `K` to discard regions of code...what has happened? ;___;

**Comment by zilongshanren** [comment]
@CestDiego  I also have no idea about the breaks. I used Magit heavily... Maybe I should do some temp fixes for myself...

**Comment by TheBB** [comment]
I noticed this too. I also don't know what happened.

**Comment by kRITZCREEK** [comment]
Can confirm. It's however still working on the "unstaged changes" tab.

**Comment by synic** [comment]
I am getting this too.  It doesn't look like it's a spacemacs thing, since the git layer hasn't been changed in days.  Should we report upstream?

**Comment by TheBB** [comment]
But if that's their `K` it should be our `C-k`, no?

**Comment by CestDiego** [comment]
Then it's our evilified thing what's wrong

**Comment by syl20bnr** [comment]
I don't thing the error is caused by the evilification, this feature is "pretty well" tested so it would surprise me but it is a possibility of course.
There must be something that changed in Magit.

**Comment by myrjola** [comment]
I see the following keybindings when I have the point over a hunk:

magit-discard `<evilified-state> K`
magit-file-untrack `K`

Is `<evilified-state>` some kind of modifier? I'll see if downgrading either magit or spacemacs fixes this...

**Comment by myrjola** [comment]
Amazing work with the rollback functionality! I did a rollback and can confirm that `K` discards in magit version `20151016.2136`.

**Comment by synic** [comment]
Guess that means I can re-open my PR?

**Comment by myrjola** [comment]
The issue seems to be load order dependent. I can't figure out how to bisect this comfortably. If I load all magit files from the latest master after Spacemacs startup I can't reproduce the bug. But updating to the latest magit package makes the issue appear.

**Comment by myrjola** [comment]
I give up :disappointed:. I couldn't even get the melpa recipe to build magit without giving errors.

```
In toplevel form:
git-rebase.el:70:1:Error: Symbol's value as variable is void: seq

In toplevel form:
magit-apply.el:33:1:Error: Symbol's value as variable is void: seq

etc.
```

My bisecting plan was to use the [MELPA recipe](https://github.com/milkypostman/melpa/blob/master/recipes/magit) to create packages (`C-c C-c` shortcut when editing the file) from different commits by using the `:commit`-option. Install the package, restarting spacemacs, rinse and repeat.

@synic I think you could open your PR. It's at least an easy solution to the problem.

**Comment by justbur** [comment]
> Is <evilified-state> some kind of modifier?

It's the way evil keybindings tied to particular states show up in keymaps. That *should* mean that the key binding will be in effect when you're in evilified state.

**Comment by justbur** [comment]
There does seem to be something wrong with the evilified state at least in the magit-status buffer. On an unstaged file, I get that `K` and `C-k` are both bound to `magit-file-untrack`, when I believe `K` should be bound to `magit-discard`. Here is my `magit-unstaged-section-map`

```
magit-unstaged-section-map
--------------------------

For more information check the manuals.

Keymap for the `unstaged' section.

key             binding
---             -------

s               magit-stage
u               magit-unstage
<evilified-state>               Prefix Command
<remap>         Prefix Command

<evilified-state> C-b           evil-scroll-page-up
<evilified-state> C-d           evil-scroll-down
<evilified-state> C-f           evil-scroll-page-down
<evilified-state> C-u           evil-scroll-up
<evilified-state> C-z           evil-emacs-state
<evilified-state> SPC           Prefix Command
<evilified-state> /             evil-search-forward
<evilified-state> :             evil-ex
<evilified-state> G             evil-goto-line
<evilified-state> N             evil-search-previous
<evilified-state> V             evil-visual-line
<evilified-state> g             Prefix Command
<evilified-state> h             evil-backward-char
<evilified-state> j             evil-next-visual-line
<evilified-state> k             evil-previous-visual-line
<evilified-state> l             evil-forward-char
<evilified-state> n             evil-search-next
<evilified-state> v             evil-visual-char

<remap> <magit-delete-thing>    magit-discard
<remap> <magit-visit-thing>     magit-diff-unstaged
```

and here's the definition of the `magit-unstaged-section-map`

```
(defvar magit-unstaged-section-map
  (let ((map (make-sparse-keymap)))
    (define-key map [remap magit-visit-thing]  'magit-diff-unstaged)
    (define-key map [remap magit-delete-thing] 'magit-discard)
    (define-key map "s" 'magit-stage)
    (define-key map "u" 'magit-unstage)
    map)
  "Keymap for the `unstaged' section.")
```
Perhaps something is wrong when the keys are defined through `remap`

By the way there's an easy way to disable evilified state temporarily. Use `C-z` to put yourself in emacs state. You can use this to narrow down the problem

**Comment by justbur** [comment]
Maybe relevant https://github.com/magit/magit/commit/cc92034cf76ba4f355b203741cfa9d71c6f99088

**Comment by myrjola** [comment]
@justbur that commit is very relevant. Good find! I copied over the magit files to my elpa directory from the commit and verified that `K` was for `magit-file-untrack`. Then I copied over the files from the commit's parent https://github.com/magit/magit/commit/a57f010ce63c4dc5012baa3927e413490567cbfa and `K` was `magit-discard`. I guess we need some development to the evilification code to fix this issue.

**Comment by justbur** [comment]
@syl20bnr I found the basic problem, but it wasn't easily solvable in the evilified state code. The issue is that `K` is bound to `magit-file-untrack` in `magit-file-section-map` but `k` is not bound at all (`k` passes through to the `magit-mode-map` it seems). This means the `K` is not automatically moved when the `magit-file-section-map` is evilified, and because it stays there it masks the new `K` in the `magit-mode-map`. 

You can check that this is the case by commenting out this line https://github.com/magit/magit/blob/14285f5ef6c451edabbddf6a3b6fe888afa33165/lisp/magit-diff.el#L1215.

**Comment by Evidlo** [comment]
I get behavior this, too.

**Comment by justbur** [comment]
Fixed in #3578
